projects/aca-content/src/lib/components/search/search-save/list/ui-list/saved-searches-list.ui-component.ts

```diff
@@ -57,6 +57,14 @@ export class SavedSearchesListUiComponent extends DataTableSchema implements Aft
   @Output()
   savedSearchOrderChanged = new EventEmitter<{ previousIndex: number; currentIndex: number }>();
 
+  @HostListener('document:keydown.escape', ['$event'])
+  onEscapeKeydown() {
+    const contextMenu = document.querySelector<HTMLElement>('.adf-context-menu');
+    if (contextMenu) {
+      document.querySelector<HTMLElement>('.adf-context-menu-source')?.focus();
```

Comment:
I guess that element is inside datatable component? So there are other ways to get it without checking whole document. You can use ViewChild to get html element of datatable document and then you can call querySelector on datatable html element to find that menu source inside. It will have better performance because you will check only part of html instead of starting from first element in document.

Reply:
@nikita-web-ua is that the same like reason like https://github.com/Alfresco/alfresco-content-app/pull/4906#discussion_r2559707283 or can we change it here?